### PR TITLE
Implement `actx.np.zeros`

### PR DESCRIPTION
**Pull request by inducer**
I'm plotting to deprecate/remove `actx.zeros` in #190. This prepares for that.

### CODE REVIEW

File: arraycontext/impl/pyopencl/fake_numpy.py [hunk 1]
```diff
@@ -39,12 +39,15 @@
...
 from arraycontext.fake_numpy import BaseFakeNumpyLinalgNamespace
+from arraycontext.impl.pyopencl.taggable_cl_array import TaggableCLArray
 from arraycontext.loopy import LoopyBasedFakeNumpyNamespace
 
 
 try:
     import pyopencl as cl  # noqa: F401
     import pyopencl.array as cl_array
+
+    from arraycontext.impl.pyopencl.taggable_cl_array import TaggableCLArray
```
Comment (alexfikl) [inline arraycontext/impl/pyopencl/fake_numpy.py:50]: `TaggableCLArray` seems to be imported twice here?

Comment (inducer) [reply arraycontext/impl/pyopencl/fake_numpy.py]: Good catch, fixed!

**Comment by inducer** [comment]
Un-auto-merging here, to give you a chance/more time to voice an opinion...

File: arraycontext/context.py [hunk 2]
```diff
@@ -306,22 +303,23 @@ def empty(self,
...
-    @abstractmethod
     def zeros(self,
               shape: Union[int, Tuple[int, ...]],
               dtype: "np.dtype[Any]") -> Array:
-        pass
+        warn(f"{type(self).__name__}.zeros is deprecated and will stop "
+            "working in 2025. Use actx.np.zeros instead.",
+            DeprecationWarning, stacklevel=2)
+
+        return self.np.zeros(shape, dtype)
```
[the remark this answers is not in the document]
Comment (alexfikl) [reply arraycontext/context.py:313]: Sounds good!

**Comment by inducer** [comment]
Thanks for taking a look!

**Pull request merged**